doc/dev/osd_internals/manifest.rst

```diff
@@ -89,6 +131,41 @@ to be wired up to provide feature parity:
   would need to be modified to instead flush and evict using
   manifests.
 
+Plans
+ 
+* Rework the tiering agent for manifest objects
+  - flush method (passive vs active)
+
+    Current flush operation for manifest objects uses a active flush model like
+    read the cotent - generating fingerprint - write and set.
+    But, exising tiering method uses copy-get and copy-from to flush the object.
+    I think the active flush model has two advantages compared to the existing method.
+    First, it can reduce the number of operations for a communication. Such as
+    Active: read -> generating fingerprint -> write and set
+    Passive: request copy_from -> copy_get -> read -> getnerating fingerprint -> write and set
+    Second, By managing I/O related to flush on the base tier, 
+    we can control overall I/Os on the cluster appropriately.
+
+  - tier agent
+
+    Because the active flush model is used, the tier agent does more jobs than before
+    (read - generating fingerprint - write and set vs sending copy_from). 
+    So, my concern is a parallism and performance.
+    since a single agent thread is probably hard to achieve high throughput.
+    To this end, the initial idea is decoupling the control and data thread.
+    Existing agent thread (agent_work) manages the list of dirty objects (pgbackend->list_partial)
+    , and flush threads do a flush job in parallel. 
```

Comment:
Why not just have N agent threads working on N pgs?

Reply:
That is also fine to me.  If each agent thread manage N pgs, it also can resolve the throughput problem.

Reply:
Right.  I'm not totally against the flush worker concept, but I suspect that partitioning by pg will be both simpler and more performant (as it will avoid lock contention on pg data structures).